app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,28 @@ module.exports = function(userDao) {
     });
   };
 
+  const findByUsername = function(username, password) {
+    return new Promise((resolve, reject) => {
+      _userDao.findByUsername(username)
+        .then((user) => {
+          return isMatchingPassword(password, user.password)
```

Comment:
I think this is getting confused with the other PR. You'll want to resolve with the user since you're going to want to store a reference to the user in the session, and your route implementation expects a user https://github.com/sendjmoon/Blogger/pull/10/files#diff-b7f406a1cfec652b44ec85434625d171R31.

There are different ways to go about this. One would be to continue the chain by having the promise resolve with an object that contains both a reference to the user and the result of `isMatchingPassword`.

Another is to just nest this.
```javascript
.then((user) => {
  isMatchingPassword(password, user.password)
    .then((isMatching) => {
      if (isMatching) {
        resolve(user);
      } else {
        reject();
      }
    });
})
```